Examples/Scripts/compareRootFiles.C

```diff
@@ -546,3 +552,48 @@ int compareRootFiles(std::string file1, std::string file2,
   std::cout << "* Input files are equal, event order aside!" << std::endl;
   return 0;
 }
+
+#ifndef __CLING__
+int main(int argc, char* argv[]) {
+  std::string file1;
+  std::string file2;
+  bool dumpDataOnFailure = false;
+  bool skipUnsupportedBranches = false;
+
+  std::vector<std::string> args(argv + 1, argv + argc);
+
+  if (args.size() < 2) {
+    std::cerr << "Usage: " << argv[0]
+              << " file1 file2 [--dump-data-on-failure] "
+                 "[--skip-unsupported-branches]\n";
+    return 1;
+  }
+
+  file1 = args[0];
+  file2 = args[1];
+
+  for (size_t i = 2; i < args.size(); ++i) {
+    if (args[i] == "--dump-data-on-failure") {
+      dumpDataOnFailure = true;
+    } else if (args[i] == "--skip-unsupported-branches") {
+      skipUnsupportedBranches = true;
+    } else {
+      std::cerr << "Unknown option: " << args[i] << "\n";
+      return 1;
+    }
+  }
```

Comment:
```suggestion
  std::vector<std::string> args(argv + 1, argv + argc);

  if (args.size() < 2) {
    std::cerr << "Usage: " << argv[0]
              << " file1 file2 [--dump-data-on-failure] "
                 "[--skip-unsupported-branches]\n";
    return 1;
  }

  bool dumpDataOnFailure = false;
  bool skipUnsupportedBranches = false;

  for (size_t i = 2; i < args.size(); ++i) {
    if (args[i] == "--dump-data-on-failure") {
      dumpDataOnFailure = true;
    } else if (args[i] == "--skip-unsupported-branches") {
      skipUnsupportedBranches = true;
    } else {
      std::cerr << "Unknown option: " << args[i] << "\n";
      return 1;
    }
  }

  const std::string file1 = args[0];
  const std::string file2 = args[1];
```

Reply:
Yes, the diff in this view is not so detailed.

- I would move the `string`-initialisation to the assignment (and make them const)
- create the `bool`s right before the are needed

Reply:
I'm inclined to keep this as is. Is that ok?